internal/archive/detect.go

```diff
@@ -0,0 +1,89 @@
+package archive
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gabriel-vasile/mimetype"
+)
+
+func detectFileType(file string) (mimeType string, err error) {
+
+	fDescriptor, err := os.Open(file)
+	if err != nil {
+		return
+	}
+	defer fDescriptor.Close()
+	// We only have to pass the file header = first 261 bytes
+	head := make([]byte, 261)
+
+	_, err = fDescriptor.Read(head)
+	if err != nil {
+		return
+	}
+
+	mtype := mimetype.Detect(head)
+	mimeType = mtype.String()
+
+	return
+}
+
+func FileIsGZ(file string) (isGZ bool, err error) {
+	mtype, err := detectFileType(file)
+	if err != nil {
+		return
+	}
+	isGZ = mtype == "application/gzip"
+	return
+}
+
+func FileIsTar(file string) (isTar bool, err error) {
+	mtype, err := detectFileType(file)
+	if err != nil {
+		return
+	}
+	isTar = mtype == "application/x-tar"
+	return
+}
+
+func DetectTGZAndUnpack(filePath string, targetDir string) (err error) {
```

Comment:
This works, but a cool idea would be to have a recursive `Unpack` function instead, with a parameter to limit the recursion depth (and a default value).

```
Unpack(filepath, targetDir string) {
  switch TYPE {
    case GZ: {
      file = Gunzip()
      Unpack(file, targetDir)
    }
    case Tar: {
      file = Untar()
      Unpack(file, targetDir)
    }
  }
}
```